### ISSUE TITLE
unbundle default Mapbox styles from SDKs

### ISSUE THREAD

**Issue by jfirebaugh** [mapbox/mapbox-gl-native#2239]
Load these via the Mapbox Styles API instead. Bundling doesn't win us anything, because network access is still needed for sources, sprite, fonts, and tiles. Unbundling allows us to remove the styles submodule from this repository (and not replace it with an npm dependency).

**Comment by incanus** [comment]
We should replace this active code path with tests for the `asset://` method of style loading, to ensure we don't have regressions there. Anyone using custom styles is likely going to be using that mechanism, even if it has online dependencies.

**Comment by ljbade** [comment]
Yes lets drop the bundled styles, and instead include a test custom style to use with `asset://` URLs.

**Comment by friedbunny** [comment]
If we ditch bundled styles, we should be sure that our caching is bulletproof. After downloading a style once, users shouldn't find themselves having to wait to re-download it — even if the style has been changed.

**Comment by friedbunny** [comment]
Unbundling styles also introduces the very real possibility that an app compiled with an older version of our library won't be able to handle a new version of the same style — e.g., the `default_marker` changes that are in the process of being introduced are all taking place within the same `streets-v8` style.

If we were fetching the current network style of `streets-v8`, markers would be broken in Android and would have been broken in iOS until last week.

**Comment by jfirebaugh** [comment]
The intent of style spec versioning is to control when breaking changes in the styles get rolled out, and ensure that we update renderers in lockstep. Although removing default markers was in fact [part of the v8 rollout](https://github.com/mapbox/gl-internal/issues/268), we were lax in making the necessary mapbox-gl-native changes in sync with this rollout. It was in fact the style pin that allowed this laxness, and unpinning will help ensure that we don't let ourselves do something like that again in the future.

**Comment by ljbade** [comment]
@jfirebaugh Now #2506 is fixed, this change is unblocked again.

**Comment by ljbade** [comment]
Is there a list of the `mapbox://` URLs we will use here?

**Comment by friedbunny** [comment]
All of the bundled styles should be available at `mapbox://styles/mapbox/style-v8`.

**Comment by ljbade** [comment]
Started work on this in `2239-no-bundled-styles` branch

**Comment by ljbade** [comment]
In my branch I have swapped all the `asset://` URLs with the `mapbox://` ones and removed the styles repo. I also removed all the styles bundling bits from the build.

I tested that this is working on Android.

Remaining:
- test on OSX
- test on iOS
- test on Linux
- add tests that use a custom `asset://` style on each platform
- update any examples code and docs in this repo
- update any examples code and docs in android-sdk and ios-sdk on mapbox.com
